modules/shared/src/main/scala/io/constellationnetwork/currency/schema/currency.scala

```diff
@@ -162,6 +167,7 @@ object currency {
     epochProgress: EpochProgress,
     dataApplication: Option[DataApplicationPart] = None,
     messages: Option[SortedSet[Signed[CurrencyMessage]]] = None,
+    globalSnapshotSyncs: Option[SortedSet[Signed[GlobalSnapshotSync]]] = None,
```

Comment:
I think that the default value should be an empty set of events 🤔 this is what you'd set as a default value with no `Option`. an `Option` here is just to ensure backward compatibility

Reply:
Don't we need backward compatibility there anyway? 🤔

Reply:
I mean, yes. The type should still remain the same. but the default value should be an empty sorted set rather than None. Consider it like:

```scala
sealed trait Wrapper[A]
case class ImplementedField[A](a: A) extends Wrapper[A]
case class NotImplementedField extends Wrapper[Nothing]
```

then, your type is:
```scala
globalSnapshotSyncs: Wrapper[SortedSet[Signed[GlobalSnapshotSync]] = ImplementedField(SortedSet.empty)
```

where for the previous global snapshots (backward compatibility) it would be:
```
.toGlobalSnapshotInfo(..., NotImplementedField)
```

Instead of this custom solution, we do use `Option` that is isomorphic to this

Reply:
Ah, you are right!